src/templates/Practice4.tsx

```diff
@@ -0,0 +1,132 @@
+import { useForm} from 'react-hook-form'
+import { object, string,InferType } from 'yup'
+import { yupResolver } from '@hookform/resolvers/yup'
+import { useState} from 'react';
+import Button from '../components/Button'
+
+  const validationSchema = object().shape({
+    postalCode: string()
+      .required('必須項目です')
+      .matches(/^\d{7}$/, '7桁の数字を入力してください'),
+    prefecture: string().required('必須項目です'),
+    city: string()
+      .required('必須項目です'),
+  }).required();
+
+type FormData = InferType<typeof validationSchema>;
+
+const Practice4: React.FC = () => {
+  const [canSubmit, setCanSubmit] = useState<Boolean>(true);
+  const [canSearch, setCanSearch] = useState<Boolean>(true);
+  const {
+    register,
+    handleSubmit,
+    formState: { errors }, getValues,setValue,trigger,
+  } = useForm<FormData>({
+    resolver: yupResolver(validationSchema),
+    defaultValues: {
+      postalCode: '',
+      prefecture: '',
+      city: '',
+    },
+    mode: 'onChange',
+  });
+
+  // Form submission handler
+  const onSubmit = handleSubmit(data => {
+    console.log(data);
+  });    
+
+  const onSearch = async() => 
+    {
+      setValue("prefecture",'');
+      setValue("city",'');
```

Comment:
Why do you need to clear these inputs? I don't think we need to clear them even when the address is not found.

Reply:
I added it so that when the user changes the postal code these fields display blank values first and then fetched address, but I have made the changes now.